include/dkm.hpp

```diff
@@ -207,6 +207,21 @@ std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> kmeans_lloyd(
 	return std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>>(means, clusters);
 }
 
+
+template <typename T, size_t N>
+size_t predict(std::vector<std::array<T, N>>& centroids, const std::array<T, N> query) {
```

Comment:
`centroids` can be declared with `const`: `const std::vector<std::array<T, N>>& centroids`. Also `query` should be a reference to avoid copying the array.